Lib/test/test_dataclasses.py

```diff
@@ -1300,6 +1300,21 @@ class D:
         self.assertTrue(is_dataclass(d.d))
         self.assertFalse(is_dataclass(d.e))
 
+    def test_is_dataclass_when_getattr_always_returns(self):
+        # See bpo-37868.
+        class A:
+            def __getattr__(self, key):
+                return 0
+
+        self.assertFalse(is_dataclass(A()))
+        self.assertFalse(is_dataclass(A))
+
+        # class A above is really testing the same things as:
+        class B:
+            __dataclass_fields__ = []
+
+        self.assertFalse(is_dataclass(B()))
+
```

Comment:
Not sure what I was thinking here. This is not a good test, and only passed because of the typo I used to have it in. I've deleted it.